samza-kafka/src/main/java/org/apache/samza/system/kafka/SamzaKafkaSystemAdmin.java

```diff
@@ -0,0 +1,680 @@
+    // Some of the functinality of this class may actually be used BEFORE start() is called.
+    // The SamzaContainer gets metadata (using this class) in SamzaContainer.apply, but this "start" actually gets called in SamzaContainer.run.
+    // Also we assume that start is called only once.
+    if (metadataConsumer == null) {
+      throw new SamzaException("Cannot start SamzaKafkaSystemAdmin with null metadataConsumer");
+    }
+
+    if (adminClient == null) {
+      throw new SamzaException("Cannot start SamzaKafkaSystemAdmin with null adminClient");
+    }
+  }
+
+  @Override
+  public void stop() {
+    if (metadataConsumer != null) {
+      metadataConsumer.close();
+    } else {
+      LOG.warn("In Stop KafkaSystemAdmin: metadataConsumer is null");
+    }
+    metadataConsumer = null;
+    if (adminClient != null) {
+      adminClient.close();
+    } else {
+      LOG.warn("In Stop KafkaSystemAdmin: adminClient is null");
+    }
+    adminClient = null;
+  }
+
+  @VisibleForTesting
+  /**
+   *
+   * Replication factor for the Changelog topic in kafka
+   * Kafka properties to be used during the Changelog topic creation
+   *
+   */
+  SamzaKafkaSystemAdmin(String systemName, Supplier<Consumer<K, V>> metadataConsumerSupplier,
+      Supplier<ZkUtils> connectZk, Supplier<AdminClient> connectAdminClient,
+      Map<String, ChangelogInfo> changelogTopicMetaInformation, Map<String, Properties> intermediateStreamProperties,
+      Properties coordinatorStreamProperties, int coordinatorStreamReplicationFactor,
+      boolean deleteCommittedMessages) {
+    this.systemName = systemName;
+    this.metadataConsumer = metadataConsumerSupplier.get();
+    this.adminClient = connectAdminClient.get();
+    this.changelogTopicMetaInformation = changelogTopicMetaInformation;
+    this.intermediateStreamProperties = intermediateStreamProperties;
+    this.coordinatorStreamProperties = coordinatorStreamProperties;
+    this.coordinatorStreamReplicationFactor = coordinatorStreamReplicationFactor;
+    this.connectZk = connectZk;
+    this.connectAdminClient = connectAdminClient;
+    this.deleteCommittedMessages = deleteCommittedMessages;
+
+    // TODO move to org.apache.kafka.clients.admin.AdminClien from the kafka.admin.AdminClient
+    // when deleteMessagesBefore is available there. Then we can get rid of this KafkaSystemAdminUtilScala
+    kafkaAdminUtils = new KafkaSystemAdminUtilsScala(systemName);
+  }
+
+  public static SamzaKafkaSystemAdmin getKafkaSystemAdmin(final String systemName, final Config config,
+      final String idPrefix) {
+
+    boolean zkSecure = false; // needs to be added to the argument if ever true is possible
+
+    // admin requires adminClient for deleteBeforeMessages
+    Properties props = new Properties();
+    String brokerListString =
+        config.get(String.format("systems.%s.consumer.%s", systemName, ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG));
```

Comment:
Should  use either `ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG` or `CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG` consistently here.